infrastructure/comms/src/connection/peer_connection/runner.rs

```diff
@@ -0,0 +1,125 @@
+//  Copyright 2019 The Tari Project
+//
+//  Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+//  following conditions are met:
+//
+//  1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+//  disclaimer.
+//
+//  2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+//  following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+//  3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+//  products derived from this software without specific prior written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+//  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+//  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+//  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+//  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+//  USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use std::{
+    sync::mpsc::{sync_channel, RecvTimeoutError, SyncSender},
+    thread,
+    time::Duration,
+};
+
+use crate::connection::{
+    connection::Connection,
+    message::Frame,
+    peer_connection::{control::ControlMessage, PeerConnectionContext, PeerConnectionError},
+    types::Direction,
+    ConnectionError,
+};
+
+macro_rules! try_recv {
+    ($inner: expr) => {
+        match $inner {
+            Ok(f) => Some(f),
+            Err(e) => match e {
+                ConnectionError::Timeout => None,
+                _ => break Err(e),
+            },
+        }
+    };
+}
+
+/// Take the given PeerConnectionContext and use it to start a worker thread which:
+/// - Establishes a connection to peer
+/// - Establishes a connection to the message consumer
+/// - Receives and handles ControlMessages
+///
+/// # Arguments
+///
+/// `context` - The PeerConnectionContext which will be owned by the worker thread
+pub(super) fn start_thread(context: PeerConnectionContext) -> SyncSender<ControlMessage> {
```

Comment:
Should this not be called "start" from our previous IRL discussions to keep it consistent with the other modules

Reply:
Perhaps, but our discussions were with a function which is in the context of a struct (in this case, the worker `my_worker.start()`) - this is a stand-alone function, `runner::start_thread()` so I thought it's a bit better to include "thread" for clarity

Reply:
What do you think?